src/librustc_codegen_ssa/mir/mod.rs

```diff
@@ -667,18 +668,25 @@ fn arg_local_refs<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>>(
                                 .zip(tys)
                                 .enumerate()
                                 .filter_map(move |(i, (decl, ty))| {
-                                    let ty = fx.monomorphize(&ty);
-                                    decl.name.map(|name| {
-                                        (variant_idx, i, name, false, ty)
-                                })
+                                    if let Some(name) = decl.name {
+                                        let ty = fx.monomorphize(&ty);
+                                        let (var_scope, var_span) = fx.debug_loc(mir::SourceInfo {
+                                            span: decl.source_info.span,
+                                            scope: decl.visibility_scope,
+                                        });
+                                        let var_scope = var_scope.unwrap_or(scope);
+                                        Some((variant_idx, i, name, false, ty, var_scope, var_span))
```

Comment:
This didn't have the desired effect of making locals live across suspension points only showing up in their source scopes. It's possible that without other changes, preserving the source visibility scopes is just wrong. Does anyone know?

Reply:
I wouldn't worry about it until we have proper MIR debuginfo, TBH.

Reply:
Left as-is for now.